website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+            new LinkObject('collection', { path : 'users/me/nodes/', query : { 'related_counts' : true, 'page[size]'  : 12, 'embed' : 'contributors' }, systemCollection : 'nodes'}, 'All My Projects'),
+            new LinkObject('collection', { path : 'users/me/registrations/', query : { 'related_counts' : true, 'page[size]'  : 12, 'embed' : 'contributors'}, systemCollection : 'registrations'}, 'All My Registrations')
+        ]);
+        // Load collection list
+        var loadCollections = function _loadCollections (url){
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig});
+            promise.then(function(result){
+                result.data.forEach(function(node){
+                    var count = node.relationships.linked_nodes.links.related.meta.count;
+                    self.collections().push(new LinkObject('collection', { path : 'collections/' + node.id + '/linked_nodes/', query : { 'related_counts' : true, 'embed' : 'contributors' }, systemCollection : false, node : node, count : m.prop(count) }, node.attributes.title));
+                });
+                if(result.links.next){
+                    loadCollections(result.links.next);
+                }
+            });
+            promise.then(self.calculateTotalPages());
+        };
+        self.init = function _collectionsInit (element, isInit) {
+            var collectionsUrl = $osf.apiV2Url('collections/', { query : {'related_counts' : true, 'page[size]' : self.pageSize(), 'sort' : 'date_created', 'embed' : 'node_links'}});
+            loadCollections(collectionsUrl);
+            args.activeFilter(self.collections()[0]);
+
+            $(window).click(function(event){
+                var target = $(event.target);
+                if(!target.hasClass('collectionMenu') && !target.hasClass('fa-ellipsis-v') && target.parents('.collection').length === 0) {
+                    self.showCollectionMenu(false);
+                    m.redraw(); // we have to force redraw here
+                }
+            });
+        };
+
+        self.addCollection = function _addCollection () {
+            var url = $osf.apiV2Url('collections/', {});
+            var data = {
+                'data': {
+                    'type': 'collections',
+                    'attributes': {
+                        'title': self.newCollectionName(),
+                    }
+                }
+            };
+            var promise = m.request({method : 'POST', url : url, config : xhrconfig, data : data});
+            promise.then(function(result){
+                var node = result.data;
+                var count = node.relationships.linked_nodes.links.related.meta.count || 0;
+                self.collections().push(new LinkObject('collection', { path : 'collections/' + node.id + '/linked_nodes/', query : { 'related_counts' : true }, systemCollection : false, node : node, count : m.prop(count) }, node.attributes.title));
+                args.sidebarInit();
```

Comment:
I don't see `sidebarInit` being passed as an arg anywhere; I would expect this to throw an error.